Builds/CMake/CMakeFuncs.cmake

```diff
@@ -524,6 +533,15 @@ macro(setup_build_boilerplate)
       add_compile_options(
         -Wno-redeclared-class-member -Wno-mismatched-tags -Wno-deprecated-register)
       add_definitions(-DBOOST_ASIO_HAS_STD_ARRAY)
+
+      # use ldd linker if available
+      execute_process(
+        COMMAND ${CMAKE_C_COMPILER} -fuse-ld=lld -Wl,--version
+        ERROR_QUIET OUTPUT_VARIABLE LD_VERSION)
+      if ("${LD_VERSION}" MATCHES "LLD")
+        append_flags(CMAKE_EXE_LINKER_FLAGS -fuse-ld=lld)
+      endif ()
+      unset(LD_VERSION)
```

Comment:
this doesn't seem to work on macos (it's not fatal, but the alternative linker is not detected). I'll investigate separately if the llvm linker can be used on macos, but that should not hold up this PR.